model/k8s/metrics.yaml

```
@@ -63,6 +63,19 @@ groups:
       - ref: network.interface.name
       - ref: network.io.direction
 
+  # k8s.container.* metrics
+  - id: metric.k8s.container.status.state
+    type: metric
+    metric_name: k8s.container.status.state
+    stability: experimental
+    brief: ""
```

```suggestion
    brief: "Describes number of K8s containers that are currently in a given state for a specific reason."
```

I wonder though if we should split `status.state` and `status.reason` to 2 different metrics. I would like to get more opinions from @open-telemetry/semconv-k8s-approvers on this.

---

Yeah, I think these should be 2 different metrics. For a given state, a k8s container can go through different reasons. For example, the `waiting` state can have a reason of `ErrImagePull`, which might transition to `ImagePullBackOff` after retries fail. If we have reason as a separate metric, users can visualize reason transitions over time.

---

@povilasv Anything I can do to help move this along. I would love to see this push forward as I have a whole engineering teams looking for this as we are moving from DataDog to Grafana and leveraging otel

---

I guess the `k8s.container.status.reason` should not be emitted when there is no `reason`?